engine/access/rpc/backend/backend_stream_transactions.go

```diff
@@ -0,0 +1,183 @@
+package backend
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/state"
+
+	"github.com/onflow/flow-go/engine/common/rpc/convert"
+	"github.com/onflow/flow-go/module/counters"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/engine/access/subscription"
+	"github.com/onflow/flow-go/engine/common/rpc"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+)
+
+// backendSubscribeTransactions handles transaction subscriptions.
+type backendSubscribeTransactions struct {
+	txLocalDataProvider TransactionsLocalDataProvider
+	executionResults    storage.ExecutionResults
+	log                 zerolog.Logger
+	broadcaster         *engine.Broadcaster
+	sendTimeout         time.Duration
+	responseLimit       float64
+	sendBufferSize      int
+
+	blockTracker subscription.BlockTracker
+}
+
+// TransactionSubscriptionMetadata holds data representing the status state for each transaction subscription.
+type TransactionSubscriptionMetadata struct {
+	txID               flow.Identifier
+	txReferenceBlockID flow.Identifier
+	messageIndex       counters.StrictMonotonousCounter
+	blockWithTx        *flow.Header
+	txExecuted         bool
+}
+
+// SubscribeTransactionStatuses subscribes to transaction status changes starting from the transaction reference block ID.
+// Expected errors:
+// - storage.ErrNotFound if a block referenced by the transaction does not exist.
+// - irrecoverable.Exception if there is an internal error related to state inconsistency.
```

Comment:
```suggestion
// - All other errors are potential symptoms of bugs or state corruption.
```

`irrecoverable` is never an expected return value.